JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeViewModel.kt

```diff
@@ -55,17 +100,24 @@ class HomeViewModel(
     private val postsRepository: PostsRepository
 ) : ViewModel() {
 
+    private val viewModelState = MutableStateFlow(HomeViewModelState(isLoading = true))
+
     // UI state exposed to the UI
-    private val _uiState = MutableStateFlow(HomeUiState(loading = true))
-    val uiState: StateFlow<HomeUiState> = _uiState.asStateFlow()
+    val uiState = viewModelState
+        .map { it.toUiState() }
```

Comment:
What about using `distinctUntilChanged()`? Is this needed, or is the `MutableStateFlow` always going to cover that case?

Reply:
Yup, `MutableStateFlow` has a `distinctUntilChanged` built in. So no need to add one separately in this mapping.